### PR TITLE
GS: Pre-round/truncate STQ values based on hardware tests

### PR DESCRIPTION
**Pull request by refractionpcsx2**
### Description of Changes
Pre-rounds and truncates STQ float values for textures on sprites.

### Rationale behind Changes
Extracted as safe code to use from #6553 so we can at least get the manhunt fixes in for the lighting.
We did some tests on hardware, and we found the truncating of float values gets pretty crazy depending on the difference between values. What is done in this PR seems to work pretty well for a bunch of games, however hardware mode won't feel all the fixes, I believe due to the difference in how GPU's handle texture coordinates.

### Suggested Testing Steps
Test any games including the ones listed below, make sure nothing looks busted in software and hardware.

All the following results are on Software mode

Ben 10:
Master:
image
PR:
image

Armored Core 2:
Master:
image
PR:
image

Beyond Good & Evil:
Master:
image
PR:
image

G-Force:  (kind of subjective but certainly not bad, matches some other lines on the scene)
Master:
image
PR:
image

Manhunt:
Master:
image
PR:
image

Shinobido - Way of the Ninja (this isn't right either, tbh, but neither is master, and the PS2 is janky as well)
Master:
image
PR:
image
PS2 for reference:
image

Persona 3:
Master:
image
PR:
image

Persona 3 again:
Master:
image
PR:
image

### CODE REVIEW

File: pcsx2/GS/GSState.cpp [hunk 1]
```diff
@@ -1656,6 +1671,40 @@ void GSState::FlushPrim()
...
+		{
+			if (!m_env.PRIM.FST) // STQ's
+			{
+				// ST's have the lowest 9 bits rounding down (from hardware tests)
+				for (u32 i = 0; i < m_vertex.tail; i++)
+				{
+					// Don't worry, I hate me too. If there's a modern way to do bit manipulation on floats, I'd love to hear it!
+					GSVertex* v = &m_vertex.buff[i];
+					u32 S = *(u32*)&v->ST.S;
+					u32 T = *(u32*)&v->ST.T;
```
Comment (stenzek) [inline pcsx2/GS/GSState.cpp:1688]:
```suggestion
					u32 T = std::bit_cast<u32>(v->ST.T);
```

(and `#include <bit>` at the top)

**Comment by stenzek** [comment]
Probably should do some performance testing, this per-vertex loop looks quite heavy. Vectorizing it could be.. tricky.

File: pcsx2/GS/GSState.cpp (hunk 1 above)
Comment (refractionpcsx2) [reply pcsx2/GS/GSState.cpp:1688]: Thank you <3

**Comment by refractionpcsx2** [comment]
For the moment I've limited this to sprites and flat draws only to try and avoid performance problems.

We do plan to move this to the vertex shaders + Software JIT directly where we should be able to do it with more performance, at that point the ghost rider shadow will work correctly.

**Review by JordanTheToaster — APPROVED** [review]

**Review by Mrlinkwii — APPROVED** [review]

**Pull request merged**